bin/drop_filter_results.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env python3
+
+import argparse
+import pandas as pd
+import pyreadr
+
+GENE_PANEL_HEADER = ["chromosome", "gene_start", "gene_stop", "hgnc_id", "hgnc_symbol"]
+GENE_PANEL_COLUMNS_TO_KEEP = ["hgnc_symbol", "hgnc_id"]
+
+
+def get_top_hits(id: str, df_results_family_AE: pd) -> pd:
+    """
+    Filter results to get only those with the id provided.
+    If there are >= 20 hits it will output all of them.
+    If there are less than 20 hits, it will output the 20 with the lowest p-value.
+    """
+    df_id: pd = df_results_family_AE[df_results_family_AE["sampleID"] == id]
+    if sum(df_id["aberrant"] == "True") >= 20:
+        return df_id
+    df_id = df_id.sort_values(by=["pValue"]).reset_index()
+    return df_id[:20]
+
+
+def annotate_with_hgnc(df_family_AE_top_hits: pd, out_drop_gene_name: str) -> pd:
+    """
+    Annotate results from DROP AE with hgnc ids.
+    """
+    df_genes = pd.read_csv(out_drop_gene_name)
+    df_genes.rename(columns={"gene_name": "hgncSymbol"}, inplace=True)
+    return df_genes.merge(df_family_AE_top_hits, left_on="gene_id", right_on="geneID")
+
+
+def keep_only_hits_in_gene_panel(df_family_top_hits: pd, gene_panel: str, module_name: str) -> pd:
+    """
+    Filter out from results any gene that is not present in the provided gene panel.
+    """
+    if gene_panel != "None":
+        gene_panel_header = GENE_PANEL_HEADER
+        df_panel = pd.read_csv(gene_panel, sep="\t", names=gene_panel_header, header=None, comment="#", index_col=False)
+        df_clinical = df_family_top_hits.loc[df_family_top_hits["hgncSymbol"].isin(df_panel["hgnc_symbol"])]
```

Comment:
this assignation is being overwritten by the next line
```suggestion
```

Reply:
Thanks for noticing, it should have been df_family_top_hits